tests/testcases.py

```diff
@@ -118,24 +124,28 @@ def stdout(self):
         clear_stdout was called."""
         return sys.stdout.getvalue()[self.stdout_start:].strip()
 
-    @patch('dusty.commands.utils.exec_docker')
+    def exec_docker_patch(self, *args):
+        args = ['/usr/local/bin/docker'] + [a for a in args]
+        self.exec_docker_process = subprocess.Popen(args=args, stdout=subprocess.PIPE, env=get_docker_env())
+
     @patch('sys.exit')
-    def run_command(self, args, fake_exit, fake_exec_docker):
+    def run_command(self, args, fake_exit):
         """Run a command through the Dusty client entrypoint, e.g. simulating
         the Dusty CLI as close as possible without having to call a subprocess.
         This command raises if the command fails, otherwise it returns the
         stdout generated by the command."""
-        self.fake_exec_docker = fake_exec_docker
-        sys.argv = ['dusty'] + args.split(' ')
-        client_entrypoint()
-        for call in fake_exit.mock_calls:
-            name, args, kwargs = call
-            if len(args) == 1 and args[0] > 0:
-                self._clear_stdout()
-                raise CommandError('Command {} returned with exit code: {}'.format(' '.join(sys.argv), args[0]))
-        result = self.stdout
-        self._clear_stdout()
-        return result
+        with patch('dusty.commands.utils.exec_docker', wraps=self.exec_docker_patch) as fake_exec_docker:
+            self.fake_exec_docker = fake_exec_docker
```

Comment:
see the helper function `assertExecDocker`, it get's used in the logs tests that are arguably unit tests